### PR TITLE
feat: change descriptor

### PR DESCRIPTION
**Pull request by matthiasdebernardini**

### CODE REVIEW

**Comment by ValuedMammal** [comment]
I got the example to run without issue, so that's good.

**Review by ValuedMammal — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -12,14 +12,21 @@ name = "async_wallet_bdk_sqlx"
...
+rustls = "0.23.12"
+tracing = "0.1.40"
+tracing-subscriber = { version = "0.3.18", features = ["env-filter", "serde_json", "json"] }
+
 
 [dependencies.bdk_wallet]
-path = "../bdk/crates/wallet"
-#git = "https://github.com/bitcoindevkit/bdk"
+#path = "../bdk/crates/wallet"
+git = "https://github.com/bitcoindevkit/bdk"
```
[the remark this answers is not in the document]
Comment (matthiasdebernardini) [reply Cargo.toml:28]: will do

**Comment by matthiasdebernardini** [comment]
I've found bugs regarding the change descriptor. I'm fixing them now and I'll add more tests.

There is also the issue of the constructor and migrations. 

One issue I'm running into is having multiple migrations from different projects.

What works for me, is managing my own migrations and not letting the crate manage that on its own, but I imagine someone thats in a similar situation to me (but isn't as familiar) would want the migrations managed for them. The other way around this is to give each crate (that uses pg) its own db. This can work but its not ideal.

For now, I have constructors for scenarios where you want to create a Store with  existing pg connection or one without.

```rust
impl Store {
    /// Construct a new [`Store`] with an existing pg connection
    #[tracing::instrument]
    pub async fn new(
        pool: Arc<Mutex<Pool<Postgres>>>,
        wallet_name: Option<String>,
        migration: Option<String>,
    ) -> Result<Self, BdkSqlxError> {
        info!("new store");
        let wallet_name = wallet_name.unwrap_or_else(|| "bdk_pg_wallet".to_string());

        Ok(Self {
            pool,
            wallet_name,
            migration,
        })
    }

    /// Construct a new [`Store`] without an existing pg connection
    #[tracing::instrument]
    pub async fn new_with_url(
        url: String,
        wallet_name: Option<String>,
    ) -> Result<Store, BdkSqlxError> {
        info!("new store with url");

        let pool = PgPool::connect(url.as_str()).await?;
        let pool = Arc::new(Mutex::new(pool));
        let wallet_name = wallet_name.unwrap_or_else(|| "bdk_pg_wallet".to_string());

        Ok(Self {
            pool,
            wallet_name,
            migration: None,
        })
    }
}
```

**Comment by ValuedMammal** [comment]
I think it's ok to have the optional migration path as you laid out.

Another idea: we already expect the database to exist for example at DATABASE_URL - would it hinder usability at all to just remove the migration step here and assume the schema is up to date?

**Comment by matthiasdebernardini** [comment]
I think its the method, because the data is stored in pg

[src/lib.rs:145:9] &row = Some(
    PgRow {
        network: "signet",
        internal_descriptor: "tr(tpubD6NzVbkrYhZ4WgCeJid2Zds24zATB58r1q1qTLMuApUxZUxzETADNTeP6SvZKSsXs4qhvFAC21GFjXHwgxAcDtZqzzj8JMpsFDgqyjSJHGa/1/*)#fd68k0ut",
        internal_last_revealed: 0,
        external_descriptor: "tr(tpubD6NzVbkrYhZ4WgCeJid2Zds24zATB58r1q1qTLMuApUxZUxzETADNTeP6SvZKSsXs4qhvFAC21GFjXHwgxAcDtZqzzj8JMpsFDgqyjSJHGa/0/*)#celxt6vn",
        external_last_revealed: 0,
    },
)

**Comment by ValuedMammal** [comment]
> Sounds good
> 
> I'm still having issues getting the test to pass.
> 
> ```rust
> assert_eq!(
>     wallet.spk_index().last_revealed_indices(), 
>     wallet_spk_index.last_revealed_indices()
> );
> ```
> 
> this snippet of code here seems to not work;
> 
> ```rust
>   assertion `left == right` failed
>     left: {External: 0, Internal: 0}
>    right: {External: 0}
> in src/tests.rs:92
> ```

My first guess is to either call `reveal_next_address` for both keychains in the test or make the last_revealed column nullable in the keychain table - right now it's set to DEFAULT 0

**Comment by matthiasdebernardini** [comment]
that fixed it, I would say this could be merged now

**Comment by ValuedMammal** [comment]
ACK

**Pull request merged**